From 632494bf78f7b70d264eed2fd9b0bf9e9045f09c Mon Sep 17 00:00:00 2001
From: Adam Williamson <awilliam@redhat.com>
Date: Fri, 1 Nov 2019 09:30:29 -0700
Subject: [PATCH] Handle NoModuleException in dnf_context_reset_modules

Signed-off-by: Adam Williamson <awilliam@redhat.com>
---
 libdnf/dnf-context.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/libdnf/dnf-context.cpp b/libdnf/dnf-context.cpp
index 47a68ff4..381c1c04 100644
--- a/libdnf/dnf-context.cpp
+++ b/libdnf/dnf-context.cpp
@@ -2559,7 +2559,12 @@ dnf_context_reset_modules(DnfContext * context, DnfSack * sack, const char ** mo
         return TRUE;
     }
     for (const char ** names = module_names; *names != NULL; ++names) {
-        container->reset(*names);
+        try {
+            container->reset(*names);
+        } catch (libdnf::ModulePackageContainer::NoModuleException & exception) {
+            g_set_error(error, DNF_ERROR, DNF_ERROR_FAILED, "%s", exception.what());
+            return FALSE;
+        }
     }
     container->save();
     if (sack != nullptr) {
-- 
2.24.0.rc1

